tic-tac-toe/src/Board.jsx

```diff
@@ -0,0 +1,70 @@
+import React, {useContext} from 'react';
+import Square from './Square';
+import BoardState from './contexts/BoardContext';
+import declareWinner from './contexts/declareWinner';
+
+const Board = (props) => {
+  const { turn, setTurn, boardState, setBoardState, history, setHistory} = useContext(BoardState);
+
+  const winner = declareWinner(boardState);
+  let status;
+  if (winner) {
+    status = `The winner is ${winner}!`;
+  }
+
+  function handleClick(num) {
+    if (!declareWinner(boardState) && !boardState[num]) {
+      const updatedHistory = [...history];
+      const updatedBoard = [...boardState];
+      updatedBoard[num] = turn;
+      updatedHistory.push(updatedBoard);
```

Comment:
Another option for storing the history is to store a collection of the turns taken as opposed to the entire board. Something like `[{turn: 1, move: 'X', square: 4}, {turn: 2, move: 'O', square: 0}]` or something like that. That way, you don't have to track the entire board - sometimes that can be a little bit easier to reason about.